config/templates/cluster-deployment.yaml

```diff
@@ -45,6 +45,30 @@ objects:
     awsAccessKeyId: ${AWS_ACCESS_KEY_ID}
     awsSecretAccessKey: ${AWS_SECRET_ACCESS_KEY}
 
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: ${CLUSTER_NAME}-pull-secret
+  type: kubernetes.io/dockercfg
+  stringData:
+    ".dockercfg": "${PULL_SECRET}"
```

Comment:
The . here makes me a little nervous for bugs, but could we rename to something more like the others, and non-docker specific. (even pull-secret)

Reply:
Afaik, this is the standard key name for a secret of type "kubernetes.io/dockercfg" (take a look at service account secrets in your cluster). If .dockercfg is not used, I would most likely get a validation error.